framework/src/userobject/SolutionUserObject.C

```diff
@@ -271,24 +271,34 @@ SolutionUserObject::readExodusII()
       if (std::find(all_elemental.begin(), all_elemental.end(), var_name) != all_elemental.end())
         _elemental_variables.push_back(var_name);
       if (std::find(all_scalar.begin(), all_scalar.end(), var_name) != all_scalar.end())
-        _scalar_variables.push_back(var_name);
+        // Check if the scalar matches any field variables, and ignore the var if it does. This
+        // means its a Postprocessor.
+        if (std::find(begin(_nodal_variables), end(_nodal_variables), var_name) ==
```

Comment:
you need to check both nodal and elemental field variables before adding the scalar variable

Reply:
is it actually possible for the user to specify an elemental variable that could trigger this? Previously we were just checking _nl and _aux for this.

Reply:
elemental variables can be in _nl and _aux in the same way that nodal variables are. 
some docs here, indirectly applies
https://mooseframework.inl.gov/syntax/AuxKernels/#2040a820-b5ba-4d14-a585-589f6e4f38bc

Reply:
I think this still needs to change to consider elemental variables.

same change as below though

Reply:
Ah, whoops -- I implemented it in one half of the if-else, below around line 289, but forgot to put it here. I'll push it up